presto-main/src/main/java/com/facebook/presto/server/ResourceManagerResource.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server;
+
+import com.facebook.presto.execution.resourceGroups.ResourceGroupManager;
+import com.facebook.presto.spi.resourceGroups.ResourceGroupInfo;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+
+import java.util.List;
+
+import static com.facebook.presto.util.ImmutableCollectors.toImmutableList;
+import static java.util.Objects.requireNonNull;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+
+@Path("/v1/resource-manager")
+public class ResourceManagerResource
+{
+    private final ResourceGroupManager resourceGroupManager;
+
+    @Inject
+    public ResourceManagerResource(ResourceGroupManager resourceGroupManager)
+    {
+        this.resourceGroupManager = requireNonNull(resourceGroupManager, "resourceGroupManager is null");
+    }
+
+    @GET
+    @Produces(APPLICATION_JSON)
+    public List<ResourceGroupInfo> getGroups()
+    {
+        return resourceGroupManager
```

Comment:
It's more canonical to write this as
```java
return resourceGroupManager.getRootResourceGroups().stream()
        .map(ResourceGroup::getInfo)
        .collect(toImmutableList());
```
Put `.stream()` on the first line and use method reference